languages/csharp/AppEncryption/AppEncryption.IntegrationTests/Regression/MetastoreMock.cs

```diff
@@ -15,17 +15,30 @@ public static class MetastoreMock
     {
         private static readonly AeadEnvelopeCrypto Crypto = new BouncyAes256GcmCrypto();
 
-         internal static Mock<MemoryPersistenceImpl<JObject>> CreateMetastoreMock(
+        internal static Mock<IMetastorePersistence<JObject>> CreateMetastoreMock(
             AppEncryptionPartition appEncryptionPartition,
             KeyManagementService kms,
             KeyState metaIK,
             KeyState metaSK,
-            CryptoKeyHolder cryptoKeyHolder)
+            CryptoKeyHolder cryptoKeyHolder,
+            IMetastorePersistence<JObject> metaStore)
         {
-            // TODO Change this to generate a mock dynamically based on the Metastore type
-            Mock<MemoryPersistenceImpl<JObject>> metastorePersistenceSpy = new Mock<MemoryPersistenceImpl<JObject>> { CallBase = true };
             CryptoKey systemKey = cryptoKeyHolder.SystemKey;
 
+            Mock<IMetastorePersistence<JObject>> metaStorePersistenceSpy = new Mock<IMetastorePersistence<JObject>>();
+
+            metaStorePersistenceSpy
+                .Setup(x => x.Load(It.IsAny<string>(), It.IsAny<DateTimeOffset>()))
+                .Returns<string, DateTimeOffset>(metaStore.Load);
+
+            metaStorePersistenceSpy
+                .Setup(x => x.LoadLatestValue(It.IsAny<string>()))
+                .Returns<string>(metaStore.LoadLatestValue);
+
```

Comment:
nit: probably don't need newline since these setup calls can be grouped together logically